### ISSUE TITLE
Test failed: JIT\Directed\tailcall\tailcall\tailcall.cmd

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#27658]
**Job:**
[coreclr-jitstress #20191102.1](https://dev.azure.com/dnceng/public/_build/results?buildId=413500)

**One of detail links:**
https://helix.dot.net/api/2019-06-17/jobs/ebc8aed0-f278-403d-9172-a61305a4aa29/workitems/JIT.Directed/console

**OS & Arch:**
windows x64
linux x64

**Mode:**
COMPlus_TieredCompilation=0
COMPlus_JitStress=2

**Note:**
I try to repro this failure, but failed due to stack overflow:
```
BEGIN EXECUTION                                                                                                          "C:\Users\vbibu\Tests_Windows_NT_x64_checked\Tests\Core_Root\corerun.exe" tailcall.dll                                 Test Start                                                                                                              
Stack overflow.                                                                                                         
Expected: 100                                                                                                           
Actual: -1073741571                                                                                                    
 END EXECUTION - FAILED                                                                                                  
FAILED
```

**Log:**
```
JIT\Directed\tailcall\tailcall\tailcall.cmd [FAIL]
      Stack overflow.
      
      Return code:      1
      Raw output file:      C:\dotnetbuild\work\ebc8aed0-f278-403d-9172-a61305a4aa29\Work\3bf09038-d5ab-4b3f-93c9-dc59f6bb0f53\Exec\JIT\Directed\Reports\JIT.Directed\tailcall\tailcall\tailcall.output.txt
      Raw output:
      BEGIN EXECUTION
       "C:\dotnetbuild\work\ebc8aed0-f278-403d-9172-a61305a4aa29\Payload\corerun.exe" tailcall.dll 
      Test Start
      Expected: 100
      Actual: -1073741571
      END EXECUTION - FAILED
      FAILED
      Test Harness Exitcode is : 1
```

category:testing
theme:tail-call
skill-level:intermediate
cost:medium

**Comment by VincentBu** [comment]
**Job:**
[coreclr-jitstress #20191110.1 Run Test Pri1 Windows_NT x64 checked](https://dev.azure.com/dnceng/public/_build/results?buildId=422326)

**Detail:**
https://helix.dot.net/api/2019-06-17/jobs/51e914b6-a3dc-403b-9bd6-9ec4a8ae320a/workitems/JIT.Directed/console

**OS & Arch:**
windows x64

**Mode:**
COMPlus_TieredCompilation=0
COMPlus_JitStress=2

**Note:**
I try to repro this failure, but failed due to stack overflow:
```
BEGIN EXECUTION                                                                                                          
"C:\Users\vbibu\Tests_Windows_NT_x64_checked\Tests\Core_Root\corerun.exe" tailcall.dll                                 
Test Start                                                                                                              
Stack overflow.                                                                                                         
Expected: 100                                                                                                           
Actual: -1073741571                                                                                                    
 END EXECUTION - FAILED                                                                                                  
FAILED
```

**Log:**
```
JIT\Directed\tailcall\tailcall\tailcall.cmd [FAIL]
      Stack overflow.
      
      Return code:      1
      Raw output file:      C:\dotnetbuild\work\51e914b6-a3dc-403b-9bd6-9ec4a8ae320a\Work\5b33a718-6919-4f2d-8f49-538fd3f89165\Exec\JIT\Directed\Reports\JIT.Directed\tailcall\tailcall\tailcall.output.txt
      Raw output:
      BEGIN EXECUTION
       "C:\dotnetbuild\work\51e914b6-a3dc-403b-9bd6-9ec4a8ae320a\Payload\corerun.exe" tailcall.dll 
      Test Start
      Expected: 100
      Actual: -1073741571
      END EXECUTION - FAILED
      FAILED
      Test Harness Exitcode is : 1
```

**Comment by VincentBu** [comment]
failed again in: [coreclr-jitstress #20191118.1](https://dev.azure.com/dnceng/public/_build/results?buildId=431634)

**Detail:**
https://helix.dot.net/api/2019-06-17/jobs/a07cd019-ba29-4486-b937-2b57191bf868/workitems/JIT.Directed/console

**Comment by AndyAyersMS** [comment]
Fixed by dotnet/runtime#1771.
